pypeit/utils.py

```diff
@@ -1476,62 +1231,45 @@ def robust_polyfit_djs(xarray, yarray, order, x2 = None, function = 'polynomial'
 
     return outmask, ct
 
-'''
-def robust_regression(x, y, ordr, outfrac, maxiter=100, function='polynomial', min=None, max=None):
-    """
-    Deprecated
-    """
-    msgs.bug("PypeIt using deprecated function")
-    msgs.error("Please contact the authors")
-    xsize=x.size
-    infrac = 1.0-outfrac
-    if infrac < 0.5: infrac = 0.5
-    slct = int(xsize*infrac)
-    if slct == xsize: slct = xsize-1
-    if ordr+1 >= slct:
-        if xsize <= 1:
-            msgs.error("At least 2 points are required for a statistical fit")
-        elif xsize == 2:
-            msgs.warn("Only a constant can be fit to 2 points")
-            msgs.info("Fitting a constant instead")
-            return func_fit(x,y,function,0)
-        elif  ordr+1 >= xsize:
-            msgs.warn("Not enough points ({0:d}) for a {1:d}th order fit".format(xsize,ordr))
-            ordr = xsize-3
-            slct = ordr+2
-            msgs.info("Changing order to a {0:d} order {1:s} fucntion".format(ordr,function))
-        else:
-            slct = ordr
-    indx = np.arange(xsize)
-    np.random.shuffle(indx)
-    ind = indx[:slct]
-    i=0
-    while True:
-        tc = func_fit(x[ind],y[ind],function,ordr)
-        diff = np.abs(y[ind]-func_val(tc,x[ind],function))
-        mad = np.median(diff)
-        w=np.argsort(diff)
-        inds=-1
-        for j in range(0,xsize-slct):
-            temp = ind[w[-1]]
-            ind[w[-1]] = indx[slct+j]
-            indx[slct+j] = temp
-            diff = np.abs(y[ind]-func_val(tc,x[ind],function))
-            if np.median(diff) < mad:
-                inds = j
-                mad = np.median(diff)
-            # Switch it back
-            temp = ind[w[-1]]
-            ind[w[-1]] = indx[slct+j]
-            indx[slct+j] = temp
-        if inds == -1 or i>maxiter: break
-        temp = ind[w[-1]]
-        ind[w[-1]] = indx[slct+inds]
-        indx[slct+inds] = temp
-        i += 1
-    return tc
-'''
+def robust_optimize(ydata, fitfunc, arg_dict, maxiter=10, inmask=None, sigma=None,invvar=None, lower=None, upper=None,
```

Comment:
doc string needed

Reply:
This is still under construction for the new method of correcting tellurics and will be documented when it becomes a functioning part of PypeIt.